micromamba/tests/test_create.py

```diff
@@ -1575,3 +1575,43 @@ def test_update_spec_list(tmp_path):
     out = helpers.create("-p", env_prefix, "-f", env_spec_file, "--dry-run")
 
     assert update_specs_list in out.replace("\r", "")
+
+
+def test_glob_in_build_string(tmp_path):
+    # Non-regression test for https://github.com/mamba-org/mamba/issues/3699
+    env_prefix = tmp_path / "test_glob_in_build_string"
+
+    pytorch_match_spec = "pytorch=2.3.1=py3.10_cuda11.8*"
+
+    # Export CONDA_OVERRIDE_GLIBC=2.17 to force the solver to use the glibc 2.17 package
+    try:
+        os.environ["CONDA_OVERRIDE_GLIBC"] = "2.17"
```

Comment:
pytest's `monkeypatch` fixture is super useful for these things:

```suggestion
        monkeypatch.setenv("CONDA_OVERRIDE_GLIBC", "2.17")
```